### ISSUE TITLE
SyntaxError: Invalid regular expression: /[\p{Lu}]/: Invalid escape

### ISSUE THREAD

**Issue by MatiasDelorenzi** [nodkz/mongodb-memory-server#587]
## Versions

- NodeJS: 14.18.1
- mongodb-memory-server-*: 8.0.4
- mongoose: 5.13.9

## What is the Problem?

When I run the tests locally, it works fine, but when I push it to gitlab the pipeline where the dependencies are installed fails giving the following error

```
.../node_modules/mongodb-memory-server-core/node_modules/camelcase/index.js:3
const UPPERCASE = /[\p{Lu}]/u;
                  ^
SyntaxError: Invalid regular expression: /[\p{Lu}]/: Invalid escape
    at Object.<anonymous> (.../node_modules/mongodb-memory-server-core/node_modules/camelcase/index.js:3:19)
```

I've tried using multiple node and mongodb-memory-server versions but it is still not working

## Do you know *why* it happenes?

*no*

**Comment by hasezoey** [comment]
what are the local nodejs version used and the gitlab nodejs version(s)? also, what is the package manager?

**Comment by MatiasDelorenzi** [comment]
The local nodejs version is the same as gitlab's, I'm using 14.18.1 and as a package manager I'm using npm

**Comment by hasezoey** [comment]
are you maybe on different architectures?
are the node_modules cached in gitlab? when yes, did you already try clearing that?

**Comment by MatiasDelorenzi** [comment]
The node_modules folder generates as the pipeline runs, so it's not cached in gitlab

**Comment by hasezoey** [comment]
> The local nodejs version is the same as gitlab's, I'm using 14.18.1 and as a package manager I'm using npm

just to confirm, you are testing on only one nodejs version, which is the same as local? (no matrix version?)

---

do you maybe use a js compiler / transpiler in-between (something like tsc or babel or ts-node) that is somehow different between local and gitlab?

---

from what i read, the error is because of regex `\p`, which is only supported (without flags) in nodejs since 10.0.0(i am unsure of the exact minor / patch version)

**Comment by MatiasDelorenzi** [comment]
I've used multiple nodejs versions (14.x, 12.x, 16.x, etc) to make sure it wasn't a problem with node. I also tried different mongodb-memory-server versions (6.x, 7.x, 8.0.4) but still throws the same error in gitlab, not locally
I'm not using any js compiler.

Maybe it has something to do with my company's gitlab configuration. I'll keep working on this and if I find something usefull I'll throw it in here!
